code/model/LDAPUtil.php

```diff
@@ -91,4 +91,16 @@ public static function little_endian($hex) {
 		return $result;
 	}
 
+	/**
+	 * Convert a date in the format of 20150323021359.0Z (YmdHis.0Z)
+	 * into Y-m-d H:i:s format in UTC.
+	 *
+	 * @param string $value
+	 * @return string
+	 */
+	public static function convert_date_to_ymd($value) {
+		$date = DateTime::createFromFormat('YmdHis.0Z', $value);
```

Comment:
I'd add a note here that LDAP uses [GeneralizedTime](http://en.wikipedia.org/wiki/GeneralizedTime) format, and that we assume it will never supply tenths-of-a-second (i.e. this is always zero) and that it will always supply it as UTC (i.e. Z at the end). At least I hope [these assertions](http://social.technet.microsoft.com/wiki/contents/articles/28222.active-directory-generalized-time-attributes.aspx) hold...